presto-common/src/main/java/com/facebook/presto/common/type/UserDefinedType.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.type;
+
+import com.facebook.presto.common.QualifiedObjectName;
+
+import static java.util.Objects.requireNonNull;
+
+public class UserDefinedType
+{
+    private final QualifiedObjectName name;
```

Comment:
Why qualified? Can it be just a simple id?

Reply:
No, all external types are managed by `FunctionNamespaceManager`, which manages catalogs. This is also aligned with the spec as `<schema-resolved user-defined type name>`.